File: server/PoultryFarmManager.Core/Models/Batch.cs
using System;
using PoultryFarmManager.Core.Enums;

namespace PoultryFarmManager.Core.Models
{
    public class Batch : DbEntity
    {
        public string Name { get; set; } = string.Empty;
        public string? Breed { get; set; }
        public BatchStatus Status { get; set; } = BatchStatus.Active;
        public DateTime StartDate { get; set; }
        public int InitialPopulation { get; set; }
        public int MaleCount { get; set; }
        public int FemaleCount { get; set; }
        public int UnsexedCount { get; set; }
        public int Population => MaleCount + FemaleCount + UnsexedCount;
        public string? Shed { get; set; }
        public string? Notes { get; set; }
        public int? DailyFeedingTimes { get; set; }
        public Guid? FeedingTableId { get; set; }
        public FeedingTable? FeedingTable { get; set; }
    }
}

File: client/webapp/src/types/batch.ts
export interface Batch {
    id: string;
    name: string;
    status: string;
    startDate: string;
    initialPopulation: number;
    maleCount: number;
    femaleCount: number;
    unsexedCount: number;
    population: number;
    breed?: string | null;
    shed?: string | null; // Optional shed/location field
    firstStatusChangeDate?: string | null;
}

export interface NewBatch {
    name: string;
    startClientDateIsoString: string;
    maleCount: number;
    femaleCount: number;
    unsexedCount: number;
    breed?: string | null;
    shed?: string | null;
    vendorId: string;
    initialCost: number;
}
